bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/TypeChecker.java

```diff
@@ -1982,26 +1982,43 @@ private static boolean checkIsLikeTupleType(Object sourceValue, BTupleType targe
         }
 
         ArrayValue source = (ArrayValue) sourceValue;
-        if (source.size() != targetType.getTupleTypes().size()) {
+        List<BType> targetTypes = new ArrayList<>(targetType.getTupleTypes());
+        int sourceTypeSize = source.size();
+        int targetTypeSize = targetTypes.size();
+        BType targetRestType = targetType.getRestType();
+
+        if (targetRestType == null && sourceTypeSize != targetTypeSize) {
+            return false;
+        }
+        if (targetRestType != null && sourceTypeSize <= targetTypeSize) {
```

Comment:
Why do we return false when sourceTypeSize == targetTypeSize ?

Reply:
The following shouldn't panic.

```ballerina
public function main() {
    [anydata, anydata] x = [1, 2];
    any a = x.cloneReadOnly();

    if !(a is [int, int, int...]) {
        panic error("didn't match");
    }
}
```

Reply:
Changed.